### ISSUE TITLE
navigate to current parsha

### ISSUE THREAD

**Issue by binyamin** [akivajgordon/tikkun.io#61]
It would be helpful if it was possible to navigate to the current week's leining with one click. This could be a button, or the default page when you enter the site. Or, both.

The OU has a zmanim API we could pull from to make this easier. (https://github.com/Orthodox-Union/zmanim-docs#to-get-parsha-info).

Sh'koach or the project!

**Comment by akivajgordon** [comment]
Thanks for opening this issue! I love the idea, and that API looks so nice and simple.

This is definitely something I've been thinking about adding for a while, but I've put it off for some reason. This shouldn't be too difficult to solve though, so I'll give it a shot in the coming days.

One thing that's always bugged me about the concept is the difference between the reading in Israel vs. outside. In some years, they can differ for several consecutive months (I think this past year, e.g.?). I don't see anything in this API that lets me specify the locale, so I'll hunt for a different one that supports both (hebcal?).

**Comment by akivajgordon** [comment]
Thinking about this more, I might reconsider this as a feature altogether, since it conflicts with a different feature that I think might be more beneficial in the long run.

Basically, I used to have a feature (no longer after a rewrite that I did) that would remember (using local storage) which section you last selected, so that the next time you visit, it will return to that same place. I feel like this is a more practical use case for what to see when visiting the app, since I assume most readers are preparing for a specific reading which doesn't correspond with the upcoming reading on the calendar (e.g. bar mitzvah, or generally preparing a few weeks in advance).

I plan to bring that feature back, so as an alternative, I think just an indicator might be helpful to show that it's the current reading. For example, as a rough idea:

![Screen Shot 2020-05-31 at 1 51 20 PM](https://user-images.githubusercontent.com/6025036/83359077-dcc21800-a345-11ea-8b0f-5b8681412a1e.png)

I don't think that would be the final design, or maybe not even close, but the point is we can indicate the current reading without it being the default when visiting.

**Comment by binyamin** [comment]
I agree that having a small indicator of the parsha would be fine.

**Comment by akivajgordon** [comment]
Another iteration of the design...

I'm not really loving it, but would like to get some feedback.

![Screen Shot 2020-06-07 at 1 17 12 AM](https://user-images.githubusercontent.com/6025036/83960985-c859b000-a85c-11ea-8547-66f6765f6485.png)

**Comment by binyamin** [comment]
Maybe "coming up" should be an expandable section, or an icon button in the header-bar. I like the idea of showing multiple weeks, but there may be a better place for it.

**Comment by akivajgordon** [comment]
@b3u Could you elaborate on the advantages of having the "Coming up" section be expandable? It seems to me like an extra click and makes the feature itself less discoverable.

I think the placement of it here makes sense because it's where all the other jump-to-parsha stuff is. Having it in the header bar would, to me, clutter that area for a seldom-used item (as in, only a select number of readers will even care about the coming-up parsha, and for those that do, they'd presumably press it only once per session).

**Comment by binyamin** [comment]
@akivajgordon 

> @b3u Could you elaborate on the advantages of having the "Coming up" section be expandable? It seems to me like an extra click and makes the feature itself less discoverable.

When you said  ([here](https://github.com/akivajgordon/tikkun.io/issues/61#issuecomment-640159084)) that you're <q>not really loving</q> the design, I had assumed you were concerned that the "Coming up" section was pushing down the main navigation section. What is your concern with the mockup you uploaded? (I happen to prefer the first version. Less obstructive.)

> they'd presumably press it only once per session).

To solve this, we could just use the current parsha as the default location. A UI indication would be optional. Those wishing to navigate to another "coming soon" parsha could navigate to it quickly.

**Comment by binyamin** [comment]
@akivajgordon Most of the UI is right-to-left, since much of the text is hebrew. The parshiyos in the Coming-soon section are left-to-right. This may confuse users.

**Comment by akivajgordon** [comment]
@b3u Thanks for the feedback! I'm totally with you – I went through the same thought process as well as I was implementing. Here was my breakdown:

I can eliminate the issue entirely by making it vertical, but it chews up a lot of vertical real estate for that kind of thing, so I didn't really like that option.

Most of the UI is in Hebrew because the actual Torah text is Hebrew of course, and the rest is basically picking the parasha, and I chose to use the Hebrew names. And since I used Hebrew names, the layout of the browsable section goes from right to left since it feels more natural that way. So I agree that the rest of the layout definitely feels RTL.

But I still feel that the majority of users are used to reading items naturally from left to right, and I didn't want there to be confusion about it the other way. I figured at least the dates would help, since you can see which is the earlier date.

In summary, I don't necessarily love either LTR, RTL, or vertically, since they each have issues. I'm hoping at least that it's a _learnability_ problem rather than a lasting _usability_ problem – once you learn it one or two times, it will not be a problem in future visits.

Your feedback is super helpful, and I encourage others to share feedback on this as well so I can consider making adjustments if it's really too painful.
